esphome/components/sml/sml_parser.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include <string>
+#include <vector>
+#include "constants.h"
+
+namespace esphome {
+namespace sml {
+
+using bytes = std::vector<uint8_t>;
+
+class SmlBase {
+ public:
+  uint8_t type;
+  uint8_t length;
+  SmlBase(const bytes &buffer, unsigned int &pos);
```

Comment:
```suggestion
  SmlBase(const bytes &buffer, size_t pos);
```

(and no need for the reference from the looks of it)